library/runtime-api/src/commonMain/kotlin/io/matthewnelson/kmp/tor/runtime/api/key/Key.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 Matthew Nelson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+@file:Suppress("EXPECT_ACTUAL_CLASSIFIERS_ARE_IN_BETA_WARNING")
+
+package io.matthewnelson.kmp.tor.runtime.api.key
+
+import io.matthewnelson.encoding.base16.Base16
+import io.matthewnelson.encoding.base32.Base32
+import io.matthewnelson.encoding.base64.Base64
+import io.matthewnelson.kmp.tor.runtime.api.Destroyable
+
+/**
+ * Base abstraction for Public/Private keys
+ * */
+public expect sealed class Key private constructor() {
+
+    public abstract fun algorithm(): String
+    public abstract fun encoded(): ByteArray?
```

Comment:
Need to re-think this returning `null`. The `Jvm` implementation is extending `java.security.Key`. Maybe for the `Key.Private` use `encodedOrThrow` with a non-null return type?